tests/eth_fixtures.py

```diff
@@ -1,16 +1,33 @@
+import os
 import pytest
+import tempfile
 
 from nkms.blockchain.eth.agents import NuCypherKMSTokenAgent, MinerAgent
 from nkms.blockchain.eth.agents import PolicyAgent
 from nkms.blockchain.eth.chains import TheBlockchain, TesterBlockchain
 from nkms.blockchain.eth.deployers import PolicyManagerDeployer, NuCypherKMSTokenDeployer
 from nkms.blockchain.eth.utilities import MockMinerEscrowDeployer
+from nkms.blockchain.eth.interfaces import Registrar, Provider
 
 
 @pytest.fixture(scope='session')
-def chain():
+def tester_registrar():
+    _, filepath = tempfile.mkstemp()
+    tester_registrar = Registrar(chain_name='tester', registrar_filepath=filepath)
+    yield tester_registrar
+    os.remove(filepath)
```

Comment:
Checked `py.test` docs - looks like indeed, the line after `yield` will execute even if an exception happened :+1: